Phonebook.TwilightSaw/Controller/ContactController.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using TwilightSaw.Phonebook.Model;
+
+namespace TwilightSaw.Phonebook.Controller;
+
+public class ContactController(AppDbContext context)
+{
+    
+    public Contact Add(Contact contact)
+    {
+        context.Contacts.Add(contact);
+        context.SaveChanges();
+        return contact;
+    }
+
+    public Contact AddToCategory(Contact contact, Category chosenCategory)
+    {
+        var category = context.Categories.FirstOrDefault(c => c == chosenCategory);
+        contact.categories = new List<Category> { category };
+        context.Contacts.Add(contact);
+        context.SaveChanges();
+        return contact;
+    }
+
+    public List<Contact> Read()
+    {
+        return context.Contacts.Select(t => t).ToList();
+    }
+
+    public List<Contact> Read(Category category)
+    {
+        return context.Contacts.Where(t => t.categories.Any(r => r.Name == category.Name)).ToList();
+    }
+
+    public void Delete(Contact contact)
+    {
+        context.Contacts.Where(c => c.Id.Equals(contact.Id)).AsNoTracking().ExecuteDelete();
+        context.SaveChanges();
+    }
+
+    public void Update(Contact contact, string updatable, string value)
```

Comment:
🟠 **Overcomplicated Code**

💡 Could you not just have done something like this?
```
public void Update(Contact contact)
{
    context.Contacts.Update(contact);
}
```
i.e. why is it important that you only update a single value?